signalfx/util.go

```diff
@@ -50,6 +50,22 @@ var ChartColorsSlice = []chartColor{
 	{"lime_green", "#6bd37e"},
 }
 
+// getPayload for viz objects is called withs: *schema.ResourceDiff or *schema.ResourceData - that's why we need this interface
+type ResourceDataAccess interface {
+	Get(key string) interface{}
+	GetOk(key string) (interface{}, bool)
+}
+
+func chartValidate(ctx context.Context, chartResource *schema.ResourceDiff, config interface{}, getPayload func(ResourceDataAccess) (*chart.CreateUpdateChartRequest, error)) error {
```

Comment:
I wonder if it is more worthwhile to abstract this further instead of passing in placed type params. 

Something like:

```go 
type ChartValidatorFunc func(input) (ChartRequest, error)

func (fn ChartValidatorFunc) Validate(ctx, resource, config) error {
  if fn == nil {
   return errors.New("no data parser provided")
  }
  payload, err := fn(resource)
  if err != nil {
    return err
  }
  return config.ValidateChart(ctx, payload)
}
```

This is something that the standard lib does with `http.HandlerFunc` as a more concrete example.